src/Simplify.cpp

```diff
@@ -4246,36 +4246,95 @@ class Simplify : public IRMutator {
     }
 
     void visit(const Shuffle *op) {
-        if (op->is_interleave()) {
-            // Mutate the vectors
-            vector<Expr> new_vectors;
-            bool changed = false;
-            for (Expr vector : op->vectors) {
-                Expr new_vector = mutate(vector);
-                if (!vector.same_as(new_vector)) {
-                    changed = true;
-                }
-                new_vectors.push_back(new_vector);
+        if (op->is_extract_element() &&
+            (op->vectors[0].as<Ramp>() ||
+             op->vectors[0].as<Broadcast>())) {
+            // Extracting a single lane of a ramp or broadcast
+            if (const Ramp *r = op->vectors[0].as<Ramp>()) {
+                expr = mutate(r->base + op->indices[0]*r->stride);
+            } else if (const Broadcast *b = op->vectors[0].as<Broadcast>()) {
+                expr = mutate(b->value);
+            } else {
+                internal_error << "Unreachable";
             }
-            int terms = (int)new_vectors.size();
+            return;
+        }
 
-            // Try to collapse an interleave of broadcast into a single broadcast.
-            const Broadcast *b1 = new_vectors[0].as<Broadcast>();
-            if (b1) {
-                bool can_collapse = true;
-                for (size_t i = 1; i < new_vectors.size() && can_collapse; i++) {
-                    if (const Broadcast *b2 = new_vectors[i].as<Broadcast>()) {
-                        Expr check = mutate(b1->value - b2->value);
-                        can_collapse &= is_zero(check);
+        // Mutate the vectors
+        vector<Expr> new_vectors;
+        bool changed = false;
+        for (Expr vector : op->vectors) {
+            Expr new_vector = mutate(vector);
+            if (!vector.same_as(new_vector)) {
+                changed = true;
+            }
+            new_vectors.push_back(new_vector);
+        }
+
+        // Only try to simplify shuffles of loads that aren't concats
+        // of vectors. This is a bit of a hacky heuristic to avoid
+        // undoing the work of AlignLoads for Hexagon.
+        if (!op->is_concat() || new_vectors[0].type().is_scalar()) {
+            // Try to convert a shuffled load into a shuffle of a load.
```

Comment:
Not that this is terribly important, but I found the language in the comment confusing. I think you are converting
shuffle({ld(ramp(a, 1, 4), ld(ramp(a+4, 1, 4), ld(ramp(a+8, 1, 4)}, {2, 3, 4, 5, 6, 7, 8}) into
ld(shuffle({ramp(a, 1, 12)},{2, 3, 4, 5, 6, 7, 8}), right?